registry/server/templates/services/templatesRepository.ts

```diff
@@ -60,22 +106,141 @@ export class TemplatesRepository {
+            name: templateName,
+        });
+        if (!templatesToUpdate.length) {
+            return { type: 'notFound' };
+        }
+        const localizedVersions = payload.localizedVersions ?? {};
+
+        await db.versioning(
+            user,
+            {
+                type: EntityTypes.templates,
+                id: templateName,
+            },
+            async (trx) => {
+                await db(Tables.Templates)
+                    .where({ name: templateName })
+                    .update({ content: payload.content })
+                    .transacting(trx);
+                await this.upsertLocalizedVersions(templateName, localizedVersions, trx);
+            },
         );
 
-        return template;
+        const updatedTemplate = await this.mustReadTemplateWithAllVersions(templateName);
+
+        return { type: 'ok', template: updatedTemplate };
     }
 
-    async upsertLocalizedVersions(templateName: string, localizedVersions: any, trx: Transaction) {
+    async upsertLocalizedVersion(
+        templateName: string,
+        locale: string,
+        localizedVersion: LocalizedVersion,
+    ): Promise<UpsertTemplateLocalizedVersionResult> {
+        const { db } = this;
+
+        const result = await db.transaction(async (trx): Promise<UpsertTemplateLocalizedVersionResult> => {
+            const existingTemplate = await trx(Tables.Templates).where({ name: templateName }).select(1).first();
+            if (!existingTemplate) {
+                return { type: 'notFound' };
+            }
+
+            const existingRecord = await trx(Tables.TemplatesLocalized)
+                .where({ templateName, locale })
+                .select(1)
+                .first();
+
+            if (existingRecord) {
+                await trx(Tables.TemplatesLocalized)
+                    .update({ content: localizedVersion.content })
+                    .where({ templateName, locale });
+            } else {
+                await trx(Tables.TemplatesLocalized).insert({
+                    templateName,
+                    locale,
+                    content: localizedVersion.content,
+                });
+            }
+
+            return { type: 'ok', localizedVersion } as const;
+        });
+
+        return result;
+    }
+
+    async deleteLocalizedVersion(templateName: string, locale: string): Promise<DeleteTemplateLocalizedVersionResult> {
+        const { db } = this;
+        const result = await db(Tables.TemplatesLocalized).where({ templateName, locale }).delete();
+        if (result === 0) {
+            return { type: 'notFound' };
+        }
+        return { type: 'ok' };
+    }
+
+    private async mustReadTemplateWithAllVersions(
+        templateName: string,
+    ): Promise<VersionedRecord<TemplateWithLocalizedVersions>> {
+        const maybeTemplate = await this.readTemplateWithAllVersions(templateName);
+        if (!maybeTemplate) {
+            throw new Error('Template not found');
```

Comment:
can you please define a custom error and add a `templateName` to the error message